packages/neon_framework/lib/src/utils/request_manager.dart

```diff
@@ -77,6 +78,7 @@ class RequestManager {
     required String cacheKey,
     required BehaviorSubject<Result<T>> subject,
     required DynamiteRawResponse<B, H> rawResponse,
+    required DynamiteSerializer<B, H> serializer,
```

Comment:
I don't like that we have to pass the serializer every time. Could we get it somehow from the raw response?

Reply:
I guess not, after the raw response has been turned into a future in a later commit.

Reply:
This is going to stay (also thinking of https://github.com/nextcloud/neon/issues/878)
I can also push the remaining PRs so you get the Idea of what I was trying to achieve.

Reply:
Just explain it in a few sentences please, opening many dependent PRs is not great.

Reply:
The idea is that the request manager will do what it's name says.
Use `http.BaseRequest` and the httpclient of the account to execute the request itself.
By executing the request itself it can directly work with the `http.StreamedResponse` to efficiently convert it into the desired `DynamiteResponse` or json for the cache.
No more need for the hard to understand/debug future juggling inside the DynamiteRawResponse. 

This requires the request manager to have access  to serializer to both deserialize the response and cache.

Reply:
Sounds good.
There is absolutely no way to avoid passing the serializer manually?

Reply:
We could later try to pass it together with the request in a record but that's kinda ugly.

Reply:
FYI this will also allow us to unify all most of the login in the different RequestManager methods.
`wrapUri` and `wrapBinary` are easily migrated to provide a `http.Request` and appropriate serializer.